tests/test_enterprise_due_diligence_gate.py

```
@@ -85,51 +86,112 @@ def test_validate_scenario_amount_accepts_positive_integer() -> None:
+            callbacks.append("strip")
+            return super().strip(*args, **kwargs)
+
+    with pytest.raises(ValueError, match="canonical lowercase full Git object identity"):
+        GATE.validate_source_commit(HostileCommit(SOURCE_COMMIT))
+
+    assert callbacks == []
 
 
 @pytest.mark.parametrize("valuation_claim", [True, "false"])
 def test_build_gate_manifest_rejects_valuation_claims(valuation_claim: object) -> None:
     with pytest.raises(ValueError, match="valuation_claim|valuation claim"):
```

Ruff (0.16.1)

[warning] Pattern passed to `match=` contains metacharacters but is neither escaped nor raw

(RUF043)